### ISSUE TITLE
Lb302 does not reset state before/after export

### ISSUE THREAD

**Issue by michaelgregorius** [LMMS/lmms#7234]
### System Information

Linux

### LMMS Version(s)

master

### Most Recent Working Version

-

### Bug Summary

Lb302 does not reset it's state before/after an export. Therefore if you export the project twice they will differ in phase.

[7223-Lb302-Project.zip](https://github.com/LMMS/lmms/files/15152125/7223-Lb302-Project.zip)

### Expected Behaviour

The exports should be identical.

### Steps To Reproduce

1. Decompress the attached file [7223-Lb302-Project.zip](https://github.com/LMMS/lmms/files/15152125/7223-Lb302-Project.zip).
2. Open the project.
3. Export the project into First.wav.
4. Export the project into Second.wav
5. Compare the files

See also the comments starting from here for more details:
https://github.com/LMMS/lmms/pull/7230#issuecomment-2081367349

### Please search the issue tracker for existing bug reports before submitting your own.

- [X] I have searched all existing issues and confirmed that this is not a duplicate.

**Comment by firewall1110** [comment]
Not only Lb302 ... :

https://github.com/user-attachments/assets/48e7db17-100a-410a-950f-73193608e702

**Is it bug ?**

**Comment by michaelgregorius** [comment]
Might be. What are the affected instruments that do not null in the two exports? Can you do isolated tests with them, i.e. not in such a complex project?

**Comment by firewall1110** [comment]
_Produce only similar output_:
Kicker, Triple Oscilator, LB302, Organic, Vibed, ZynAddSubFx
    _Some "stars":_
OpulenZ, VeSTige :: DSK Harmonica
   _Little noise shadow_:
Nescaline

  _Tested also (produce the same result)_:
Mallets, Monstro, Watsyn, PATMAN (095_Sweep_Pad), SID, Xpressivo, AUDIO FILE Processor, SF2 :: MSGM_Flute.sf2, 
LV2 :: MDA Piano
  _Tested with_ lmms-1.3.0-alpha.1.102+g89fc6c9-linux-x86_64.AppImage 

_Project_: 
[st2004_08_05.mmpz.zip](https://github.com/user-attachments/files/16490839/st2004_08_05.mmpz.zip)

**Comment by firewall1110** [comment]
And this is why I think , that it is not a bug:

https://github.com/user-attachments/assets/0b59ea3d-a22f-4dd6-b24d-e0c16cd75641

Used PRO DAW, PRO ReWire and VST soft.synths ...

**Comment by michaelgregorius** [comment]
> And this is why I think , that it is not a bug:
> prodaw_state.mp4
> 
> Used PRO DAW, PRO ReWire and VST soft.synths ...

I don't understand why it follows from that video that it is not a bug?

**Comment by firewall1110** [comment]
Yes "_Tested also (produce the same result):_ " are "good".

In last video I demonstrate, that **_the same thing is in other_** DAW&SoftSynth, that used in PRO purpose. In this case LMMS and LMMS synth are more accurate (in prodaw_state.mp4 near all instruments are "bad", and **_it depends on preset_** : except two solo (one is "good", but another "bad"), others are one 16 midi channel ReWire synth, and some (string, piano, base drums) sounds are "good" ).

Why sound is only similar?
(1) Instruments are using LFO period generator, that often not triggered by  note on, but "rotate" all the time.
(2) Instrument using some kind of randomization - "human style" imitation.

**Comment by michaelgregorius** [comment]
Yes, it might indeed be a shortcoming of some of the LMMS synths. In VSTs and other synths you often have the option to enable or disable phase reset for the oscillators and LFOs. With phase reset enabled they should more or less sound the same on each rendering when the phase reset is triggered by notes.

**Comment by firewall1110** [comment]
Quote: "_more or less sound the same_"
My opinion is that all LMMS "bad" synths _more or less sound the same_ on each rendering even now. 
So it is some kind of taste question. 

And my opinion:  if something _**even can be**_ treated as taste question - it is not bug. 
Bug is something objective - not depending on any subjective opinion. 

Now I try to formulate this as a programmer:
"_Lb302 does not reset state before/after export_" - this is some kind _convention violation_. 
Is it written in Lb302 **_specification_**?
Is **_in LMMS standards_** is written that all  LMMS synths MUST _reset state before/after export_?
If **_no_** - it is **_not bug_**. 

P.S.
My opinion is: DO NOT write something like this in any LMMS standard! 
Only after LMMS became top 10 professional DAW - than we can open discussion about this question.

**Comment by michaelgregorius** [comment]
@firewall1110, if you check comment https://github.com/LMMS/lmms/pull/7230#issuecomment-2081367349 you will find that Lb302 renders in a deterministic way on the first render. Once the render is through the instance stays in an "in between" state that IIRC also is not caused by the instruments being run all the time or something similar. So for me this has rather a bug character.

> Now I try to formulate this as a programmer: "_Lb302 does not reset state before/after export_" - this is some kind _convention violation_. Is it written in Lb302 **_specification_**? Is **_in LMMS standards_** is written that all LMMS synths MUST _reset state before/after export_? If **_no_** - it is **_not bug_**.

As a software engineer I know that specifications can be quite useful. However, to my knowledge LMMS does not really have specifications for its software components. If I follow your logic then this means that LMMS does not have any bugs, either. :wink:

> P.S. My opinion is: DO NOT write something like this in any LMMS standard! Only after LMMS became top 10 professional DAW - than we can open discussion about this question.

IMO a lot of things have to be fixed before LMMS can enter the top 10 of professional DAWs. So we should not make work on LMMS and agreeing on things without a specification dependent on that state.

**Comment by firewall1110** [comment]
Quote: "_if you check comment_ https://github.com/LMMS/lmms/pull/7230#issuecomment-2081367349 _you will find that Lb302 renders in a deterministic way on the first render. Once the render is through the instance stays in an "in between" state_"

Now I understand Your position - (Yes) this has rather a bug character. 
(**_Off topic:_** If Lb302 works  in a deterministic way, except after render: may be something is not complete _in rendering process_ - problem not in Lb302 code? Can we somehow test ("trace") what happens with synth when simply play , and when rendering happens (and compare)?)

Quote: " _If I follow your logic then this means that LMMS does not have any bugs, either._"

In this case LMMS does not have any bugs of  "_convention violation_" type. But have _functional_ bugs - in current master Lb302 has no sound at all ...  , have _regressions_ - SID has regression, I am not sure about sound, but XRun-s with Jack  ...   
Have _GUI problems_ - there are no specifications in LMMS, but we have some common conventions, how good GUI should be ...

 P.S.
Probably my English is so bad but thinking is so complicated: sometimes You  only partially understand, what I write ...

**Comment by firewall1110** [comment]
Now I export in different way:
open LMMS + project export close LMMS
open LMMS + project export close LMMS
(LMMS configured to load last project on start).

Results are:

_Produce only similar output:_ Kicker, Triple Oscilator, Organic, Vibed
_Little noise shadow:_ Nescaline

_Tested also (exact the same sound):_
Mallets, LB302, Monstro, OpulenZ, Watsyn, PATMAN (095_Sweep_Pad), SID, Xpressivo, ZynAddSubFx, AUDIO FILE Processor, SF2 :: MSGM_Flute.sf2, LV2 :: MDA Piano, VeSTige :: DSK Harmonica.

P.S.
Now LB302 became "good" ... as You wrote ...

**Comment by rubiefawn** [comment]
I am also able to replicate this.
